vm/src/pyobject.rs

```diff
@@ -639,6 +640,12 @@ where
     T: IntoPyObject,
 {
     fn get_item(&self, key: T, vm: &VirtualMachine) -> PyResult {
+        if let Ok(map) = PyMappingMethods::try_from_borrowed_object(vm, self) {
+            if let Some(getitem) = map.subscript {
+                return getitem(self.clone(), key.into_pyobject(vm), vm);
+            }
+        }
+
```

Comment:
do we still need `__getitem__` call under this line? I think PyMapping covers the whole cases

Reply:
Yes. there exists extra steps (sequence protocol check & method lookup) as [cpython implementation](https://github.com/python/cpython/blob/main/Objects/abstract.c#L180)